src/server/report/app.js

```diff
@@ -349,9 +344,13 @@ class App extends LitElement {
 			});
 		});
 
-		const selectedBrowser = browsers[this._selectedBrowserIndex] ||
-			browsers.find(b => browserResults.get(b.name) < tests.length) ||
-			browsers[0];
+		if (!browsers[this._selectedBrowserIndex]) {
+			this._selectedBrowserIndex = Math.max(
+				browsers.findIndex((b) => browserResults.get(b.name) < tests.length),
+				0
+			);
```

Comment:
We need to remember the index we automatically pick for them so that it stays as they iterate.